components/identity/org.wso2.carbon.identity.application.authentication.endpoint/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/AuthenticationEndpoint.java

```diff
@@ -78,6 +78,10 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
                 loadPage = "passivests_login.do";
             } else if (request.getParameter("type").equals("oauth2") || request.getParameter("type").equals("oidc")) {
                 loadPage = "oauth2_login.do";
+            } else if (request.getParameter("type").equals("totp")) {
+                loadPage = "totp_login.do";
+            } else if (request.getParameter("type").equals("totp_error")) {
```

Comment:
use <constant>.equals(<variable>) to avoid possible NPE. Better to use constants instead of hard coded strings